src/main/java/step4/Car.java

```diff
@@ -0,0 +1,32 @@
+package step4;
+
+public class Car {
+    private final String name;
+    private int position;
+
+    public Car(String name) {
+        this.name = name;
+        this.position = 0;
+    }
```

Comment:
> 또한 자동차의 이름을 클래스로 만들어서 이름에 대한 검증을 생성자에서 하는 것도 생각해보았습니다.
그렇게 되면 어쨋든 입력을 받아서 검증(자동차 이름 객체 생성) 후 재시도 하는 부분이 InputView 에 있을 수 밖에 없을 것 같은데요,
결국 InputView 가 자동차의 이름 객체를 반환해주는 모양이 됩니다.
InputView 는 .. 먼가 단순히 입출력만 담당해야 할 것 같은데 비즈니스 객체를 반환하게 되니 .. 먼가.. 먼가 싶은 느낌입니다.

말씀하신 대로, `InputView`에서 검증 후 `CarName` 객체를 반환하는 방식은 `InputView`의 역할을 벗어나는 행위로 보여져요.
일반적으로 `InputView`는 단순 입출력을 담당하는 것이 맞지만, 객체 생성을 위한 최소한의 검증은 필요할 수 있습니다.

`CarName` 클래스에서 검증을 수행하는 것은 도메인 로직의 응집도를 높이고, 객체의 유효성을 보장하는 좋은 방법입니다. 
다만, 이 경우 `InputView`와 `CarName` 양쪽에서 검증이 이루어지므로 중복처럼 느껴질 수 있습니다.

하지만, 두 검증은 엄연히 다른 목적을 가집니다. `InputView`의 검증은 사용자 입력의 형식적 유효성을 확인하는 것이고, `CarName`의 검증은 도메인 객체의 유효성을 확인하는 것입니다. 
예를 들어, `InputView`에서는 이름의 길이 제한이나 특수문자 포함 여부를 검증하고, `CarName`에서는 이름의 의미적 유효성을 검증할 수 있습니다.

따라서, 두 검증을 모두 수행하는 것이 반드시 중복이라고 볼 수는 없습니다. 
오히려 각 계층의 책임에 맞는 검증을 수행함으로써 코드의 안정성을 높일 수 있습니다.

물론, 팀의 컨벤션이나 프로젝트의 특성에 따라 `InputView`에서만 검증을 수행하거나, `CarName`도 같이 검증을 수행하는 것 둘다 가능하니 팀원들과 충분히 논의하여 일관된 규칙을 정하고, 그 규칙을 따르는게 중요해보여요!

```suggestion
    public Car(String name) {
        this.name = new CarName(name);
        this.position = 0;
    }
```

고민하시는 부분에 조금이나마 도움이 되셨으면 좋겠네요 🙇